orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableTask.java

```diff
@@ -55,6 +57,24 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(30);
   }
 
+  @Override
+  public long getDynamicTimeout(StageExecution stage) {
+    ManifestTimeout timeout = stage.mapTo(ManifestTimeout.class);
+    if (timeout.timeoutMinutes.isPresent()) {
+      return TimeUnit.MINUTES.toMillis(timeout.timeoutMinutes.get());
+    }
+
+    return getTimeout();
+  }
+
+  private static class ManifestTimeout {
+    @Nonnull private final Optional<Long> timeoutMinutes;
```

Comment:
The benefit here is really small, so no need to change, but in general it's often better to use `OptionalLong` instead of `Optional<Long>` as putting a `long` in an `Optional` requires first boxing it to a `Long` then putting it in the `Optional`; getting the value out requires the same two steps.  An `OptionalLong` just has the one step of directly putting the `long` in the `Optional`.  In this case, given that we're not doing this in a tight loop or anything, I think the difference is negligible so it's fine as is.